### PR TITLE
Add documentation for helm, application sources, and parameter overrides

### PR DESCRIPTION
**Pull request by jessesuen**
Docs can be read here:
https://github.com/jessesuen/argo-cd/tree/helm-docs/docs

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # ArgoCD Getting Started
 
-An example Ksonnet guestbook application is provided to demonstrates how ArgoCD works.
+An example guestbook application is provided to demonstrates how ArgoCD works.
```
Comment (alexmt) [inline docs/getting_started.md]: nitpicking: demonstrates -> demonstrate

File: docs/parameters.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+1. A team maintains a "dev" environment, which needs to be continually updated with the latest
+version of their guestbook application after every build in the tip of master. To address this use
+case, the application would expose an parameter named `image`, whose value used in the `dev`
+environment contains a placeholder value (e.g. `example/guestbook:replaceme`). The placeholder value
+would be determined externally (outside of git) such as a build systems. Then, as part of the build
+pipeline, the parameter value of the `image` would be continually updated to the freshly built image 
+(e.g. `argocd app set guestbook -p guestbook=image=example/guestbook:abcd123`). A sync operation
+would result in the application being redeployed with the new image.
+
+2. An repository of helm manifests is already public available (e.g. https://github.com/helm/charts).
```
Comment (alexmt) [inline docs/parameters.md]: An repository -> A repository

Comment (alexmt) [inline docs/parameters.md]: public available -> publicly available

File: docs/tracking_strategies.md [hunk 3]
```diff
@@ -1,41 +1,45 @@
...
-$ argocd app set guestbook -p guestbook=image=example/guestbook:abcd123
-$ argocd app sync guestbook
-```
-
-Note that in all tracking strategies, any parameter overrides set in the application instance will be honored.
+If a git commit SHA is specified, the application is effectively pinned to the manifests defined at
+the specified commit. This is the most restrictive of the techniques and is typically used to
+control production environments.
 
+ssetSince commit SHAs cannot change meaning, the only way to change the live state of an application
```
Comment (alexmt) [inline docs/tracking_strategies.md]: ssetSince -> Since

**Review by alexmt — APPROVED** [review]
Added few minor comments

**Comment by jessesuen** [comment]
Thanks for reviewing so carefully. Will make the changes.

**Comment by jessesuen** [comment]
Holding off on merging docs until release is cut.

**Pull request merged**